src/common/extensionMessaging.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as http from "http";
+
+/**
+ * Message server parameters.
+ */
+export let ServerParams = {
+    PORT: 8099,
+    HOST: "127.0.0.1"
+};
+
+/**
+ * Defines the messages sent to the extension.
+ * Add new messages to this enum.
+ */
+export enum ExtensionIncomingMessage {
+    START_PACKAGER,
+    STOP_PACKAGER
+}
+
+/**
+ * Generic interface for messages with arguments.
+ */
+export interface MessageWithArgs<T> {
+    message: T;
+    args?: any[];
```

Comment:
Instead of using an any, I think I'd be better if we defined an interface for each message type we are planning on sending , and then the type of the message is MessageType1 | MessageType2 | MessageType3 etc... (Initially we'd just need a single message type to handle executing commands from the command pallete)